src/int128_struct_impl.h

```diff
@@ -0,0 +1,133 @@
+#ifndef SECP256K1_INT128_STRUCT_IMPL_H
+#define SECP256K1_INT128_STRUCT_IMPL_H
+
+#include "int128.h"
+
+#if defined(_M_X64) | defined(_M_ARM64) | defined(_WIN64) /* MSVC */
```

Comment:
Hm, I'm not sure about `_WIN64`. This is defined by icc for example, and as I understand also by mingw. (The Internet says that mingw, which is basically "gcc for windows", will support the MSVC intrinsicts, but it will also have a native 128 bit, so we'll probably want to use the latter.)

MSVC is what provides those intrinsics and `_MSC_VER` should really only be defined on MSVC. What about one of these two?

```
#if defined(_MSC_VER) && defined(_WIN64)
```
or
```
#if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_ARM64))
```

Currently, the two versions should be equivalent. This may change if MSVC starts to support another 64 bit target in the future, and then I believe it is likely that `__mulh` und `__umulh` will exist, so I lean for the first option. But anyway, when that change happens to MSVC, we can deal with it...

Parts of my comment are guesswork so far, I believe we'll need to test compilation.

Reply:
done.